enclave/sgx/verifier.c

```diff
@@ -0,0 +1,308 @@
+                oe_result_str(result));
+
+    //verify QvE report data
+    //report_data = SHA256([nonce || quote || expiration_check_date || expiration_status || verification_result || supplemental_data]) || 32 - 0x00
+    OE_CHECK(oe_sha256_init(&sha_handle));
+
+    //nonce
+    OE_CHECK(oe_sha256_update(&sha_handle, &(p_qve_report_info->nonce), sizeof(p_qve_report_info->nonce)));
+
+    //quote
+    OE_CHECK(oe_sha256_update(&sha_handle, p_quote, quote_size));
+
+    //expiration_check_date
+    OE_CHECK(oe_sha256_update(&sha_handle, &expiration_check_date, sizeof(expiration_check_date)));
+
+
+    //collateral_expiration_status
+    OE_CHECK(oe_sha256_update(&sha_handle, &collateral_expiration_status, sizeof(collateral_expiration_status)));
+
+
+    //quote_verification_result
+     OE_CHECK(oe_sha256_update(&sha_handle, &quote_verification_result, sizeof(quote_verification_result)));
+
+    //p_supplemental_data
+    if (p_supplemental_data) {
+        OE_CHECK(oe_sha256_update(&sha_handle, p_supplemental_data, supplemental_data_size));
+
+    }
+
+    //get the hashed report_data
+    //
+    OE_CHECK(oe_sha256_final(&sha_handle, &report_data_hash));
+
+    memcpy(&report_data, &report_data_hash, sizeof(report_data_hash));
+
+    if (memcmp(&p_qve_report->body.report_data, &report_data, sizeof(report_data)) != 0) {
+        result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+        goto done;
+    }
+    
+    do {
+        //Check MiscSelect in QvE report
+        if (((p_qve_report->body.miscselect) & g_qve_misc_select_mask) != g_qve_misc_select) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+
+        //Check Attribute in QvE report
+        if (((p_qve_report->body.attributes.flags) & g_qve_attribute_flags_mask) != g_qve_attribute_flags) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+
+        if (((p_qve_report->body.attributes.xfrm) & g_qve_attribute_xfrm_mask) != g_qve_attribute_xfrm) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+        
+        //Check Mrsigner in QvE report
+        if (memcmp(&(p_qve_report->body.mrsigner), g_qve_mrsigner, sizeof(g_qve_mrsigner)) != 0) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+
+        //Check Prod ID in QvE report
+        if (p_qve_report->body.isvprodid != QVE_PRODID) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+
+        //Check QvE ISV SVN in QvE report
+        if (p_qve_report->body.isvsvn < qve_isvsvn_threshold) {
+            result = OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED;
+            break;
+        }
+
+        result = OE_OK;
+        
+    } while (0);
+
+done:
+    if (p_oe_qve_report)
+        free(p_oe_qve_report);
```

Comment:
```suggestion
    oe_free(p_oe_qve_report);
```

Reply:
Same as above, we don't need to malloc/free buffer after using `oe_verify_raw_sgx_report()` to verify report.